### DISCUSSION TITLE
Filter engineering view per Grade

### DISCUSSION THREAD

**Discussion by ternite** [iaincollins/icarus#13] [Ideas & Suggestions]
The engineering view does not allow to give you an "at a glance" view of all your engineering material, even when considering that the program filters by category (raw, manifactured, data, xeno).

While farming High Grade Emission signals, I could really use a view that only shows the materials of grade 5. Thus, I propose adding 5 filter buttons, so that you can deselect grades so that only those grades are shown that remain selected.

In addition, if there is only 1 selected category remaining, the group heading (Alloys, Capacitors, Chemical, ...) could also be made invisible. That way, it would be possible to have all Grade 5 materials in the view without the need to scroll. At a glance.

**Comment by iaincollins** [comment]
I'd really like to solve for this too.

My plan is to  give the option to 'pin' any materials, blueprints and engineers to the main panel of the Engineering screen so people can track whatever they want at a glance - and for that to persist across all terminals, after restarts, etc.

The last release introduced a new Preferences system that lets users save changes like this, so it should be useful for a feature like this.

**Comment by excalith** [comment]
Hey there, replying to an old thread to bump a few ideas into.

### Engineering Planner
Would be amazing to "Pin" required materials of a selected blueprint from its chosen materials. Just like EDOMH does. 

- A navigation button to this page just above the materials of the vertical menu could show tracked materials and their amounts which we already display in materials pages. 
- Show / Hide completed functionality would be nice here
- Maybe we can create seperate wishlists per ship and we can show them as individual lists or show all

### Blueprints Page
It shows the applied blueprints and other blueprints. However, we can already see those within Ship page  (maybe it could show the name of the applied blueprint there)

Instead, I think 
- It should filter the blueprints depending on their category with search option perhaps. In this page we can show the grade unlocked per blueprint as well.
- The blueprint sub-page with details that shows the engineers and grades, I think sorting of engineers should start from Grade 5 instead, and invited / known last. From here we could also pin the required materials for the planner I mentioned above. Rightmost of the blueprint name might have "Add to planner" that shows a modal to ask which grades to add. Also, I personally wouldn't mind not seeing if my current ship has that blueprint applied or not here as well.

### Engineers Page
Maybe we can add a filtering here as well, to quickly check which engineer can upgrade which blueprint. A sorting could be nice as well depending on Name, Grade and Distance.

---

For those ideas we already have all the data to work with except for the planner.

Planner has a great potential for further improvements, such as displaying which modules are requires the materials etc. But it would require some effort to find the best UX that Icarus applies throughout everything else. 

Would love to hear some feedback!
